### PR TITLE
Fix for download path write permission issue

### PR DESCRIPTION
**Pull request by xoriole**
Checked if there is sufficient write permission on the download path before attempting to download the torrent.

### CODE REVIEW

**Comment by xoriole** [comment]
retest this please

File: TriblerGUI/dialogs/startdownloaddialog.py [hunk 1]
```diff
@@ -147,12 +147,19 @@ def on_received_metainfo(self, metainfo):
...
+                                                                     "download", "", QFileDialog.ShowDirsOnly)
 
         if len(chosen_dir) != 0:
             self.dialog_widget.destination_input.setCurrentText(chosen_dir)
 
+        if not is_dir_writable(chosen_dir):
+            ConfirmationDialog.show_message(self.dialog_widget, "Insufficient Permissions",
+                                            "Tribler cannot download to <i>%s</i> directory. "
+                                            "Please add proper write permissions to the directory "
+                                            "or choose another download directory" % chosen_dir,
```
Comment (devos50) [inline TriblerGUI/dialogs/startdownloaddialog.py]: Please add a period after the last sentence.

File: TriblerGUI/dialogs/startdownloaddialog.py [hunk 2]
```diff
@@ -163,7 +170,15 @@ def on_download_clicked(self):
             ConfirmationDialog.show_error(self.window(), "No files selected",
                                           "Please select at least one file to download.")
         else:
-            self.button_clicked.emit(1)
+            download_dir = self.dialog_widget.destination_input.currentText()
+            if not is_dir_writable(download_dir):
+                ConfirmationDialog.show_message(self.dialog_widget, "Insufficient Permissions",
```
Comment (devos50) [inline TriblerGUI/dialogs/startdownloaddialog.py]: Please extract this string into an attribute so we can reuse it.

File: TriblerGUI/tribler_window.py [hunk 3]
```diff
@@ -444,6 +449,7 @@ def start_download_from_uri(self, uri):
         self.download_uri = uri
 
         if get_gui_setting(self.gui_settings, "ask_download_settings", True, is_bool=True):
+            print "asking download settings"
```
Comment (devos50) [inline TriblerGUI/tribler_window.py]: Don't forget to remove this print statement.

File: TriblerGUI/utilities.py [hunk 4]
```diff
@@ -204,3 +204,13 @@ def get_gui_setting(gui_settings, value, default, is_bool=False):
     if is_bool:
         val = val == True or val == 'true'
     return val
+
+
+def is_dir_writable(path):
+    try:
+        open(os.path.join(path, 'temp'), 'w')
```
Comment (devos50) [inline TriblerGUI/utilities.py]: Why don't you use `os.access`? https://docs.python.org/2/library/os.html#files-and-directories

**Pull request merged**